units/src/amount/result.rs

```diff
@@ -0,0 +1,573 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Provides a monodic result type that is used to return the result of
+//! doing mathematical operations (`core::ops`) on amount types.
+
+use core::{fmt, ops};
+
+use NumOpResult as R;
+
+use super::{Amount, SignedAmount};
+
+/// Result of an operation on [`Amount`] or [`SignedAmount`].
+///
+/// The type parameter `T` should be normally `Amout` or `SignedAmount`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[must_use]
+pub enum NumOpResult<T> {
+    /// Result of a successful mathematical operation.
+    Valid(T),
+    /// Result of an unsuccessful mathematical operation.
+    Error(NumOpError),
+}
+
+impl<T> NumOpResult<T> {
+    /// Returns the contained valid amount, consuming `self`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the result is an `Error`.
+    pub fn unwrap(self) -> T {
```

Comment:
In 4d6911b12c0dec91ebf2ba1065224300e6231ad5:

Should also have `expect` somewhere.

Reply:
Added `expect`. Added `track_caller` to all three methods that panic (`expect`, `unwrap`, `unwrap_err`)